09_maps/main.go

```diff
@@ -111,18 +136,271 @@ func main() {
 	// Print the BMI of each person in the array.
 	// Hint: Use a loop.
 
+	// create empty array of maps
+	var people []map[string]int = []map[string]int{}
+	// fill array of maps
+	people = append(people, map[string]int{"height": 180, "weight": 70})
+	people = append(people, map[string]int{"height": 160, "weight": 50})
+	people = append(people, map[string]int{"height": 170, "weight": 80})
+	people = append(people, map[string]int{"height": 150, "weight": 60})
+	people = append(people, map[string]int{"height": 190, "weight": 90})
+	// debug
+	fmt.Println(people)
+
+	var counter int = 0
+
+	for {
+		if counter >= len(people) {
+			break
+		}
+		// print bmi by calling the calcBMI with an array value of people
+		fmt.Println("BMI is", calcBMI(people[counter]))
+		counter++
+	}
+
 	// For later exercises, use the same format as the previous one.
 
-	// TODO: Create a function that takes an array of maps as an argument and
-	// prints the average BMI of the people in the array.
+	// calculate average BMI
+	fmt.Println("Avg BMI is", calcAvgBMI(people))
 
-	// TODO: Create a function that takes an array of maps as an argument and
-	// returns the map with the highest BMI.
+	// calculate highest BMI
+	fmt.Println("Highest BMI is", highestBMI(people))
 
-	// TODO: Create a function that takes an array of maps as an argument and
-	// add as a new key-value pair to each map the key "bmi" with the value
-	// being the BMI of the corresponding person.
-	// Hint: Use a loop.
-	// Hint: Create a new map for each person.
-	// Hint: Create a new array of maps.
+	// add BMI to array of maps
+	fmt.Println(addBMI(people))
+
+	// test helper int function
+	fmt.Println(largestInt([]int{}))                      // 0
+	fmt.Println(largestInt([]int{1, 2, 3}))               // 3
+	fmt.Println(largestInt([]int{-1, -2, 0}))             // 0
+	fmt.Println(largestInt([]int{100, 0, 3000, 0, -100})) // 3000
+
+	// test helper string function
+	fmt.Println(largestString([]string{"", ""}))                         // ""
+	fmt.Println(largestString([]string{"a", "bc", "def"}))               // "def"
+	fmt.Println(largestString([]string{"xyz", "xyzabcd", "qwertyuiop"})) // "qwertyuiop"
```

Comment:
your function `largestString` is still incorrect, here's a few more examples 

```suggestion
	fmt.Println(largestString([]string{"xyz", "xyzabcd", "qwertyuiop"})) // "qwertyuiop"
		fmt.Println(largestString([]string{"a", "dcd", "ef"})) // "bcd"
		fmt.Println(largestString([]string{"a", "b", "c"})) // "a"
		fmt.Println(largestString([]string{"abc", "de", "f"})) // "abc"
		fmt.Println(largestString([]string{"abc", "", "def"})) // "abc"
	  fmt.Println(largestString([]string{"", "abc", ""})) // "abc"
```

Reply:
Added